Support http_proxy env variable

Authentication fails if the openshift cluster's public endpoint is reachable only through an http proxy.

Error message on my environment: 
> 018/05/28 07:44:32 oauthproxy.go:582: error redeeming code (client:10.128.4.1:33746 ("10.0.1.152")): Post https://openshift_public_url/oauth/token:  dial tcp xx.xx.xx.xx:443: getsockopt: connection timed out
2018/05/28 07:44:32 oauthproxy.go:399: ErrorPage 500 Internal Error Internal Error

I tried configuring uppercase and lowercase environment variables http_proxy & https_proxy but it doesn't work. I looked at the authproxy.go source code and proxy handling doesn't seem implemented.

This is a real show stopper for us because we would like to expose [Prometheus](https://docs.openshift.com/container-platform/3.9/install_config/cluster_metrics.html#openshift-prometheus) dashboards to users coming from Internet and our openshift cluster can't be directly exposed on Internet.

> I would expect you use -login-url and -redeem-url to override the oauh server endpoint values in this case. Oauth Proxy is supposed to be run within the cluster, going through another proxy sounds backwards.

@simo5 : I tried setting **-redeem-url** but get the following error

```
2018/11/10 07:56:05 oauthproxy.go:635: error redeeming code (client:10.128.4.1:33902 ("xx.xx.xx.xx")): got 403 from "https://internal-openshift-master.red.act.ed/auth/token" {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"forbidden: User \"system:anonymous\" cannot post path \"/auth/token\": no RBAC policy matched","reason":"Forbidden","details":{},"code":403}
2018/11/10 07:56:05 oauthproxy.go:434: ErrorPage 500 Internal Error Internal Error
```

I get similar error in the browser if I set **-login-url**.

@s-urbaniak : http proxy environment variables are not used, PR to follow

--- a/providers/openshift/provider.go
+++ b/providers/openshift/provider.go
@@ -87,7 +87,7 @@ func (p *OpenShiftProvider) LoadDefaults(serviceAccount string, caPaths []string
 	}
 	p.hostreviews = hostreviews
 
-	if err := p.setCA(caPaths); err != nil {
+	if err := p.setCAandTransportProxy(caPaths); err != nil {
 		return nil, err
 	}
 
@@ -125,7 +125,7 @@ func (p *OpenShiftProvider) LoadDefaults(serviceAccount string, caPaths []string
 }
 
 // SetCA initializes the client used for connecting to the master.
-func (p *OpenShiftProvider) setCA(paths []string) error {
+func (p *OpenShiftProvider) setCAandTransportProxy(paths []string) error {
 	if p.Client == nil {
 		p.Client = &http.Client{
 			Jar:       http.DefaultClient.Jar,
@@ -144,6 +144,7 @@ func (p *OpenShiftProvider) setCA(paths []string) error {
 		return err
 	}
 	p.Client.Transport = &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
 		TLSClientConfig: &tls.Config{
 			RootCAs: pool,
 		},

